packages/node/src/ethereum/utils/eth-bucket.ts

```diff
@@ -23,16 +17,45 @@ export function getETHBucketAppInterface(addr: string): AppInterface {
   };
 }
 
-/// A StateChannelTransaction commitment that uses this as the terms will accept
-/// any resolution that is a resolution to ETH, with no limit on the amount of
-/// ETH returned by the resolution. The commitment to the free balance app
-/// instance uses this as the terms.
-export const unlimitedETH: Terms = {
-  assetType: 0,
-  limit: MaxUint256,
-  token: AddressZero
-};
-
 export function encodeETHBucketAppState(state: ETHBucketAppState) {
   return defaultAbiCoder.encode([ethBucketStateEncoding], [state]);
 }
+
+export const fromAppState = (
+  appState: ETHBucketAppState
+): { [s: string]: BigNumber } => {
+  const ret = {};
+  for (const { to, amount } of appState) {
+    ret[to] = bigNumberify(amount._hex);
+  }
+  return ret;
+};
+
+export function flip(a: { [s: string]: BigNumber }) {
+  const ret = {};
+  for (const key of Object.keys(a)) {
+    ret[key] = Zero.sub(a[key]);
+  }
+  return ret;
+}
+
+// todo(xuanji): check for underflow
+export function merge(
+  a: { [s: string]: BigNumber },
+  b: { [s: string]: BigNumber }
+) {
+  const ret = {};
+  for (const key of Object.keys(a)) {
+    if (b[key]) {
+      ret[key] = a[key].add(b[key]);
+    } else {
+      ret[key] = a[key];
+    }
+  }
+  for (const key of Object.keys(b)) {
+    if (!a[key]) {
+      throw Error(`mismatch ${Object.keys(a)} ${Object.keys(b)}`);
+    }
+  }
+  return ret;
+}
```

Comment:
What is any of this stuff above?

Reply:
I think it makes sense but maybe name them more explicitly? Since they are not methods on a class I would advise against super short names like `flip` and `merge`.